backend/dal/Services/Concrete/BuildingService.cs

```diff
@@ -246,16 +247,21 @@ public Building Update(Building building)
                 {
                     if (!allowEdit) throw new NotAuthorizedException("User may not add parcels to a property they don't own.");
                     var existingParcel = this.Context.Parcels
+                    .Include(p => p.Buildings)
                     .FirstOrDefault(pb => pb.Id == parcel.Id);
                     if(existingParcel != null)
                     {
+                        parcel.Buildings.Clear(); // Do not modify the list of buildings associated to parcels when performing building updates.
```

Comment:
yeah, open to suggestions here. Here is what is going on in abstract:

We don't want to trust this part (highlighted in bold) of the incoming buildings relationships building -> *parcels -> buildings*

1) Clear out all of the parcels associated buildings (since there is no setter on this list). 
2) Add associated buildings such that our incoming parcel's buildings is identical to the existing parcel's buildings in the db.

Since the incoming parcel's buildings are identical to the existing parcel's buildings no add/updates/removals should take place.